crates/polars-plan/src/logical_plan/optimizer/simplify_functions.rs

```diff
@@ -103,8 +142,105 @@ pub(super) fn optimize_functions(
                 } => Some(AExpr::Function {
                     input: input.clone(),
                     function: FunctionExpr::Boolean(BooleanFunction::IsNull),
-                    options: *options,
+                    options: options,
+                }),
+                // not(a == b) => a != b
+                AExpr::BinaryExpr {
+                    left,
+                    op: Operator::Eq,
+                    right,
+                } => Some(AExpr::BinaryExpr {
+                    left: left,
+                    op: Operator::NotEq,
+                    right: right,
+                }),
+                // not(a != b) => a == b
+                AExpr::BinaryExpr {
+                    left,
+                    op: Operator::NotEq,
+                    right,
+                } => Some(AExpr::BinaryExpr {
+                    left: left,
+                    op: Operator::Eq,
+                    right: right,
                 }),
+                // not(a < b) => a >= b
+                AExpr::BinaryExpr {
+                    left,
+                    op: Operator::Lt,
+                    right,
+                } => Some(AExpr::BinaryExpr {
+                    left: left,
+                    op: Operator::GtEq,
+                    right: right,
+                }),
+                // not(a <= b) => a > b
+                AExpr::BinaryExpr {
+                    left,
+                    op: Operator::LtEq,
+                    right,
+                } => Some(AExpr::BinaryExpr {
+                    left: left,
+                    op: Operator::Gt,
+                    right: right,
+                }),
+                // not(a > b) => a <= b
+                AExpr::BinaryExpr {
+                    left,
+                    op: Operator::Gt,
+                    right,
+                } => Some(AExpr::BinaryExpr {
+                    left: left,
+                    op: Operator::LtEq,
+                    right: right,
+                }),
+                // not(a >= b) => a < b
+                AExpr::BinaryExpr {
+                    left,
+                    op: Operator::GtEq,
+                    right,
+                } => Some(AExpr::BinaryExpr {
+                    left: left,
+                    op: Operator::Lt,
+                    right: right,
+                }),
+                #[cfg(feature = "is_between")]
+                // not(col('x').is_between(a,b)) => col('x') < a || col('x') > b
+                AExpr::Function {
+                    input,
+                    function: FunctionExpr::Boolean(BooleanFunction::IsBetween { closed }),
+                    ..
+                } => {
+                    if !matches!(expr_arena.get(input[0]), AExpr::Column(_)) {
```

Comment:
I did this because I am not sure if otherwise it would hurt performance if we duplicate a more complex input into two predicates